adapter/oam.go

```diff
@@ -261,27 +268,35 @@ func CreateComponents(scfg StaticCompConfig) error {
 	}
 	return nil
 }
-func convertOAMtoMeshmodel(def []byte, schema string, isCore bool, meshmodelname string) ([]byte, error) {
+func convertOAMtoMeshmodel(def []byte, schema string, isCore bool, meshmodelname string, mcfg MeshModelConfig) ([]byte, error) {
 	var oamdef v1alpha1.WorkloadDefinition
 	err := json.Unmarshal(def, &oamdef)
 	if err != nil {
 		return nil, err
 	}
 	var c meshmodel.ComponentDefinition
 	c.Metadata = make(map[string]interface{})
+	metaname := strings.Split(manifests.FormatToReadableString(oamdef.ObjectMeta.Name), ".")
+	var displayname string
+	if len(metaname) > 0 {
+		displayname = metaname[0]
```

Comment:
I would love for this not to be a numeric index reference, but one that is named instead in order to avoid fragility (slightly).

Reply:
The length of array is checked above so if code reaches at this line then it is guaranteed to have 0th element.

Reply:
Readability is also improved when named keys are used.

Reply:
Actually here I am just doing some string manipulation to extract Readable name from name with suffix like ".Istio". Nothing much to move into a map or struct.